src/main/scala/com/ing/wbaa/gargoyle/sts/oauth/OAuth2Authorization.scala

```diff
@@ -39,16 +39,24 @@ class OAuth2Authorization(oAuth2tokenVerifier: OAuth2TokenVerifier) {
   }
 
   /**
-   * get the token from header or cookie or parameters
+   * get the token from header or cookie or parameters or body
    *
    * @return the authorization token
    */
   private def bearerToken: Directive1[Option[String]] =
```

Comment:
For this, I'd be tempted to use a value class since `String` is a bit generic and the type system can help us. (Tokens aren't interchangeable with, for example, names. But not are strings.)

```scala
case class BearerToken(value: String) extends AnyVal

private[this] def bearerToken: Directive1[Option[BearerToken]] =
…
```

Reply:
changed